### DISCUSSION TITLE
Enum types should implement IBitwiseOperators

### DISCUSSION THREAD

**Discussion by UniqeId6542** [dotnet/runtime#100960] [Ideas]
I wanted to write an extension method:
```
public static T SetFlag<T>(this T value, T flag, bool setFlag)
    where T : struct, Enum, IBitwiseOperators<T, T, T>
{
    if (setFlag)
    {
        return value | flag;
    }
    else
    {
        return value & ~(flag);
    }
}
```
But I can't use it since any given Enum type doesn't implement IBitwiseOperators.  This, even though Enum types already do implement all of the required operators.

So, could you have the Enum types also derive from IBitwiseOperators?
Thanks

**Comment by huoyaoyuan** [comment]
Enum lacks a lot of interfaces implementations, even `IEquatable<T>`. #17456 #20008
